### PR TITLE
docs: Update SIP template

### PR DESCRIPTION
**Pull request by john-bodley**
### SUMMARY

Adding documentation regarding numbering of SIPs.

### TEST PLAN

CI.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/sip.md [hunk 1]
```diff
@@ -6,9 +6,9 @@ labels: "#SIP"
 ---
 
 *Please make sure you are familiar with the SIP process documented*
-(here)[https://github.com/apache/superset/issues/5602]
+(here)[https://github.com/apache/superset/issues/5602]. The SIP number should be the next increment from the latest SIP listed [here](https://github.com/apache/superset/issues?q=is%3Aissue+label%3Asip).
```
Comment (etr2460) [inline .github/ISSUE_TEMPLATE/sip.md:9]: i like this wording better, but i'm fine either way

```suggestion
(here)[https://github.com/apache/superset/issues/5602]. The SIP number should be the next number after the latest SIP listed [here](https://github.com/apache/superset/issues?q=is%3Aissue+label%3Asip).
```

File: .github/ISSUE_TEMPLATE/sip.md [hunk 2]
```diff
@@ -6,9 +6,9 @@ labels: "#SIP"
...
 
-## [SIP] Proposal for XXX
+## [SIP-\<number>] Proposal for <title>
```
Comment (etr2460) [inline .github/ISSUE_TEMPLATE/sip.md:11]:
```suggestion
## [SIP-<number>] Proposal for <title>
```

**Review by etr2460 — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/sip.md (hunk 2 above)
Comment (john-bodley) [reply .github/ISSUE_TEMPLATE/sip.md:11]: I needed to escape this for the markdown to render correctly. I can reexamine this.

Comment (etr2460) [reply .github/ISSUE_TEMPLATE/sip.md:11]: oh, interesting. no worries then

**Review by amitmiran137 — APPROVED** [review]

**Pull request merged**